Реализованы фичи по ТЗ 13 Спринта.

kvaker88 opened the pull request:
Добавлены следующие фичи:
1) «Отзывы».
2) «Поиск». 
3) «Общие фильмы».
4) «Рекомендации».
5) «Лента событий».
6) «Удаление фильмов и пользователей».
7) «Фильмы по режиссёрам».
8) «Популярные фильмы».

NovoseltsevVladimir committed:
Добавить функционал общие фильмы

NovoseltsevVladimir committed:
Добавить функционал общие фильмы

sofjagyy committed:
add-most-populars

sofjagyy committed:
Улучшена читабельность метода getPopularFilms

sofjagyy committed:
Merge pull request #10 from kvaker88/add-most-populars

add-most-populars

NovoseltsevVladimir committed:
Промежуточный вариант функционала с режиссерами

NovoseltsevVladimir committed:
Исправить по результатам ревью

NovoseltsevVladimir committed:
Добавить функциональность по режиссерам

kvaker88 committed:
Merge pull request #8 from kvaker88/add-common-films

Добавлен функционал "Общие фильмы"

NovoseltsevVladimir committed:
Исправить ошибки по тестам

NovoseltsevVladimir committed:
Merge branch 'develop' into add-director

NovoseltsevVladimir committed:
Исправить check-style unused import

sofjagyy committed:
add-recommendations

sofjagyy committed:
checkstyle

sofjagyy committed:
add-remove-endpoint

NovoseltsevVladimir committed:
Merge pull request #16 from kvaker88/add-director

Add director

sofjagyy committed:
Merge branch 'develop' into add-remove-endpoint

sofjagyy committed:
checkstyle FilmRepository.java

sofjagyy committed:
Update FilmRepository.java

sofjagyy committed:
checkstyle

sofjagyy committed:
checkstyle

sofjagyy committed:
checkstyle

sofjagyy committed:
merge

sofjagyy committed:
Merge pull request #21 from kvaker88/add-remove-endpoint

add-remove-endpoint

sofjagyy committed:
Merge branch 'develop' into add-recommendations

Commit:
Add-search

Commit:
Add-search

Commit:
Checkstyle

Commit:
Checkstyle

sofjagyy committed:
fix

sofjagyy committed:
Merge pull request #19 from kvaker88/add-recommendations

add-recommendations

Commit:
Merge remote-tracking branch 'origin/develop' into add-search

# Conflicts:
#	src/test/java/ru/yandex/practicum/filmorate/controller/FilmControllerTest.java
#	src/test/java/ru/yandex/practicum/filmorate/controller/UserControllerTest.java
#	src/test/java/ru/yandex/practicum/filmorate/repository/film/FilmRepositoryTest.java
#	src/test/java/ru/yandex/practicum/filmorate/repository/user/UserRepositoryTest.java
#	src/test/java/ru/yandex/practicum/filmorate/service/FilmServiceTest.java
#	src/test/java/ru/yandex/practicum/filmorate/service/UserServiceTest.java
#	src/test/java/ru/yandex/practicum/filmorate/validation/FilmValidatorTest.java
#	src/test/java/ru/yandex/practicum/filmorate/validation/FriendshipValidatorTest.java
#	src/test/java/ru/yandex/practicum/filmorate/validation/LikeValidatorTest.java
#	src/test/java/ru/yandex/practicum/filmorate/validation/UserValidatorTest.java

Commit:
Fix conflict

kvaker88 committed:
Merge pull request #28 from kvaker88/add-search

Add search

Arkady465 committed:
Add files via upload

Arkady465 committed:
Merge pull request #29 from kvaker88/add-reviews

add-reviews

Commit:
Add-feed

Commit:
Checkstyle

Commit:
Checkstyle

kvaker88 committed:
Merge pull request #30 from kvaker88/add-feed

Add-feed

sofjagyy committed:
Исправлен порядок Genre у Film

sofjagyy committed:
Исправлен порядок Genre при  create у Film

sofjagyy committed:
Дополнительные проверки у Director

sofjagyy committed:
fix 22

NovoseltsevVladimir committed:
Исправить тесты

NovoseltsevVladimir committed:
Merge pull request #31 from kvaker88/newFix

Исправить тесты

NovoseltsevVladimir committed:
Исправить check-style

NovoseltsevVladimir committed:
Merge pull request #33 from kvaker88/newFix

Исправить check-style

NovoseltsevVladimir committed:
Исправить логику добавления лайков

NovoseltsevVladimir committed:
Merge pull request #35 from kvaker88/newFix

Исправить логику добавления лайков

NovoseltsevVladimir committed:
Исправить unused imports

NovoseltsevVladimir committed:
Merge pull request #37 from kvaker88/newFix

Исправить unused imports

sofjagyy committed:
убрала лишнее, добавила аннотацию

kesch9 reviewed (commented):
Несколько комментариев к работе

sofjagyy committed:
Убраны лишние зависимости

sofjagyy committed:
Убраны неиспользуемые методы из интерфейса FilmStorage

sofjagyy committed:
Убраны неиспользуемые методы из модели Film

sofjagyy committed:
Исправлены возвращаемые значения у DirectorStorage

sofjagyy committed:
Убран лишний метод getPopularFilms

sofjagyy committed:
Убрана лишняя проверка у ReviewValidator

Commit:
Fix: FilmServiceTest

Commit:
Fix: DirectorService. Добавлено ограничение на длину имени в 255 символов.

Commit:
Fix: UserFeedService слит с UserService.
Также исправлены замечания методов из бывшего UserFeedService и добавлен новый приватный метод checkUserExists для сокращения повторов.

Commit:
Fix: getCommonFilms добавлена проверка на существование обоих пользователей.

Commit:
Fix: FilmValidator и немного ReviewValidator. Также добавлена дополнительная проверка в FilmService для создания и обновления фильма.

Commit:
Fix: FilmRepository, переписан SQL запрос для getCommonFilms. Плюс, исправил ошибку Checkstyle.

Commit:
Fix: RecommendationService переписан запрос и частично логика класса.
Также поправил FilmRepository, соединил методы подгрузки режиссёров и жанры, так как они во всех случаях используются поочерёдно.

Commit:
Checkstyle

kesch9 reviewed (approved):
Approve

Commit:
Удалены неиспользуемые методы.

Commit:
Checkstyle
